javascript/tagAutocomplete.js

```diff
@@ -479,19 +504,49 @@ onUiUpdate(function () {
     if (acConfig === null) {
         try {
             acConfig = JSON.parse(readFile("file/tags/config.json"));
+            if (acConfig.translation.onlyShowTranslation) {
+                acConfig.translation.searchByTranslation = true; // if only show translation, enable search by translation is necessary
+            }
         } catch (e) {
             console.error("Error loading config.json: " + e);
             return;
         }
     }
-    // Load main tags
+    // Load main tags and translations
     if (allTags.length === 0) {
         try {
-            allTags = loadCSV();
+            allTags = loadCSV(`file/tags/${acConfig.tagFile}`);
         } catch (e) {
             console.error("Error loading tags file: " + e);
             return;
         }
+        if (acConfig.extra.extraFile) {
+            try {
+                allTranslations = loadCSV(`file/tags/${acConfig.extra.extraFile}`);
+                if (acConfig.onlyTranslationExtraFile) {
+                    for (let i = 0; i < allTranslations.length; i++) {
+                        if (allTranslations[i][0]) {
+                            allTags[i][2] = allTranslations[i][0];
+                        }
+                    }
+                } else {
+                    allTranslations.map(x => {
+                        let i = 0
+                        for (; i < allTags.length; i++) {
+                            if (x[2] && x[0] === allTags[i][0] && x[1] === allTags[i][1]) {
+                                allTags[i][2] = x[2];
+                            }
+                        }
+                        if (i === allTags.length) {
+                            allTags.push(x);
+                        }
+                    });
+                }
+            } catch (e) {
+                console.error("Error loading extra translation file: " + e);
+                return;
+            }
+        }
```

Comment:
- Fixed the bug with the onlyTranslationExtraFile config key
- Renamed allTranslations to extras since it can be new tags as well
- Added a few comments
- Better performance for the second part (from ~100 ms to 10-20 ms on my system) & simplified code

```suggestion
        if (acConfig.extra.extraFile) {
            try {
                extras = loadCSV(`file/tags/${acConfig.extra.extraFile}`);
                if (acConfig.extra.onlyTranslationExtraFile) {
                    // This works purely on index, so it's not very robust. But a lot faster.
                    for (let i = 0, n = extras.length; i < n; i++) {
                        if (extras[i][0]) {
                            allTags[i][2] = extras[i][0];
                        }
                    }
                } else {
                    extras.forEach(e => {
                        // Check if a tag in allTags has the same name as the extra tag
                        if (tag = allTags.find(t => t[0] === e[0])) {
                            if (e[2]) // If the extra tag has a translation, add it to the tag
                                tag[2] = e[2];
                        } else {
                            // If the tag doesn't exist, add it to allTags
                            allTags.push(e);
                        }
                    });
                }
            } catch (e) {
                console.error("Error loading extra translation file: " + e);
                return;
            }
        }
```